### PR TITLE
[Release] Fix leaks in zexe rust bindings

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR updates the OCaml-rust bindings so that the OCaml garbage collector always calls the appropriate `delete` function when an object is freed. Zexe PR at o1-labs/zexe#24.

I have compiled this, but not tested it; any feedback from testing would be much appreciated.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/lib/zexe_backend/zexe_backend_common/dlog_based_proof.ml [hunk 1]
```diff
@@ -76,27 +76,27 @@ module type Inputs_intf = sig
...
-      -> Scalar_field.Vector.t
-      -> Scalar_field.Vector.t
-      -> Scalar_field.Vector.t
-      -> Scalar_field.Vector.t
-      -> Scalar_field.Vector.t
-      -> Scalar_field.Vector.t
-      -> Scalar_field.Vector.t
-      -> Scalar_field.Vector.t
-      -> Scalar_field.Vector.t
+         w:Scalar_field.Vector.t
```
Comment (imeckler) [inline src/lib/zexe_backend/zexe_backend_common/dlog_based_proof.ml:79]: nice

**Review by imeckler — COMMENTED** [review]
Looks good -- maybe we run a small testnet just to make sure there aren't any weird edge cases where we double free?

**Comment by imeckler** [comment]
Also this is unrelated but can we stop compiling with libsnark?

**Review by bkase — APPROVED** [review]

**Comment by bkase** [comment]
> Looks good -- maybe we run a small testnet just to make sure there aren't any weird edge cases where we double free?

Good call -- I'll run a sandbox instance now which should be good enough I think. (it was good enough to surface the leaks anyway)

**Comment by bkase** [comment]
It looks like the build segfaults though during snark generation at the moment

**Comment by mrmr1993** [comment]
> It looks like the build segfaults though during snark generation at the moment

@bkase I'm debugging this at the moment

**Comment by mrmr1993** [comment]
Updated, I had misplaced a prefix change so `proof_delete` was being used instead of `proof_evaluations_delete`. This was not a fun mistake to track down..

**Pull request merged**